resolver/user_mutations.resolvers_test.go

```diff
@@ -55,58 +56,92 @@ func expectInsertUser(mock sqlmock.Sqlmock, mockUser models.User) {
 		).
 		WillReturnRows(rows)
 }
-func getCreateUserTestCase() []struct {
-	name     string
-	req      fm.UserCreateInput
-	wantResp *fm.User
-	wantErr  bool
-} {
-	cases := []struct {
-		name     string
-		req      fm.UserCreateInput
-		wantResp *fm.User
-		wantErr  bool
-	}{
-		{
-			name:    ErrorFromCreateUser,
-			req:     fm.UserCreateInput{},
-			wantErr: true,
+
+type createUserType struct {
+	name      string
+	req       fm.UserCreateInput
+	wantResp  *fm.User
+	wantErr   bool
+	init      func() *gomonkey.Patches
+	initmocks func(mock sqlmock.Sqlmock, mockUser models.User)
+}
+
+func errorFromCreateUserCase() createUserType {
+	return createUserType{
+		name:    ErrorFromCreateUser,
+		req:     fm.UserCreateInput{},
+		wantErr: true,
+		init:    func() *gomonkey.Patches { return nil },
```

Comment:
lets use the same function for it. 

the SQL mocking can be done in the init function itself. Any particular reason to not do that?

Reply:
No there is no any specific reason but just to seprate patches and mocks